Automattic/simplenote-android
# Update Rename Tag

### Fix
Update the entity identification value and user interface for renaming tags.  A hashed value based on the tag name is used to uniquely identify like tags to avoid duplication.  The **_Rename Tag_** dialog interface has been updated to include error messages when a tag name is invalid (i.e. contains a space or is empty).  The **_Error_** dialog is shown when a tag was unable to be renamed for an unknown reason rather than failing silently like the current implementation.  See the screenshots below for illustration.

![update_rename_tag](https://user-images.githubusercontent.com/3827611/87073093-15250380-c1da-11ea-980a-84560544c8a3.png)

### Test
1. Open navigation drawer.
2. Tap ***EDIT*** button in ***Tags*** header.
3. Tap any tag in list.
4. Notice ***Rename Tag*** dialog is shown.
5. Add space to tag name.
6. Notice ***Tags cannot contain spaces*** error message is shown.
7. Notice ***Save*** button is disabled in dialog.
8. Delete space from tag name.
9. Notice ***Tags cannot contain spaces*** error message is hidden.
10. Notice ***Save*** button is enabled in dialog.
11. Add any alphanumeric, special, or emoji character to tag name.
12. Tap ***Save*** button in ***Rename Tag*** dialog.
13. Notice tag is renamed.
14. Tap back arrow in app bar.
15. Close navigation drawer.
16. Tap note in list with renamed tag from Step 3.
17. Notice tag is renamed.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 2b75529b with:
> Fixed a bug with duplicate tags and special characters in tag names
